activestorage/lib/active_storage/analyzer/pdf_analyzer/mu_pdf.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  # This analyzer requires the {poppler}[https://poppler.freedesktop.org/] system library, which is not provided by Rails.
+  class Analyzer::PDFAnalyzer::MuPDF < Analyzer::PDFAnalyzer
+    class << self
+      def accept?(blob)
+        super && mutool_exists?
+      end
+
+      def mutool_path
+        ActiveStorage.paths[:mutool] || "mutool"
+      end
+
+      def mutool_exists?
```

Comment:
I'm not happy with this. It already exists the the previewer, and I wanted to dry it, but was unsure where to place the code.

Reply:
@brenogazzola Let's follow the [Rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) and not refactor too early here. I think possibly some interface where you can define (system) dependencies should exist with creating hard requires.